internal/querynodev2/handlers.go

```diff
@@ -249,6 +254,68 @@ func (node *QueryNode) querySegments(ctx context.Context, req *querypb.QueryRequ
 	}, nil
 }
 
+func (node *QueryNode) optimizeSearchParams(ctx context.Context, req *querypb.SearchRequest, deleg delegator.ShardDelegator) (*querypb.SearchRequest, error) {
+	// no hook applied, just return
+	if node.queryHook == nil {
+		return req, nil
+	}
+
+	log := log.Ctx(ctx)
+
+	serializedPlan := req.GetReq().GetSerializedExprPlan()
+	// plan not found
+	if serializedPlan == nil {
+		log.Warn("serialized plan not found")
+		return req, merr.WrapErrParameterInvalid("serialized search plan", "nil")
+	}
+
+	channelNum := req.GetTotalChannelNum()
+	// not set, change to conservative channel num 1
+	if channelNum <= 0 {
+		channelNum = 1
+	}
+
+	plan := planpb.PlanNode{}
+	err := proto.Unmarshal(serializedPlan, &plan)
+	if err != nil {
+		log.Warn("failed to unmarshal plan", zap.Error(err))
+		return nil, merr.WrapErrParameterInvalid("valid serialized search plan", "no unmarshalable one", err.Error())
+	}
+
+	switch plan.GetNode().(type) {
+	case *planpb.PlanNode_VectorAnns:
+		sealed, growing := deleg.GetSegmentInfo()
```

Comment:
If we turn the ignoreGrowing flag on, I guess this would be wrong. Maybe we should add a if statement or simply always ignore growing to make the optimization safe.

Reply:
+1 to second option. Since growing segment will be brute force ant estimate counting growing one will always make the estimation inaccurate.
Fixing it